src/inmanta/server/agentmanager.py

```diff
@@ -135,10 +139,10 @@ def get_dependencies(self) -> List[str]:
         presession.add_listener(self)
 
     def new_session(self, session: protocol.Session) -> None:
-        self.add_background_task(self.register_session(session, datetime.now()))
+        self.add_background_task(self._register_session(session, datetime.now()))
 
     def expire(self, session: protocol.Session, timeout: float) -> None:
-        self.add_future(self.expire_session(session, datetime.now()))
+        self.add_background_task(self._expire_session(session, datetime.now()))
```

Comment:
This was add_future because this was one of the tasks that could be executed while already stopped. When making this a background_task it generated many teardown error when enabling logging during test execution

Reply:
seems to be resolved by two phase shutdown